pkg/virt-controller/watch/migration.go

```diff
@@ -1022,10 +1073,26 @@ func (c *MigrationController) sync(key string, migration *virtv1.VirtualMachineI
 		pod = pods[0]
 	}
 
-	vmiDeleted := vmi == nil || vmi.DeletionTimestamp != nil
-	migrationFinalizedOnVMI := vmi.Status.MigrationState != nil && vmi.Status.MigrationState.MigrationUID == migration.UID && vmi.Status.MigrationState.EndTimestamp != nil
+	if vmiDeleted := vmi == nil || vmi.DeletionTimestamp != nil; vmiDeleted {
+		return nil
+	}
+
+	if (migration.DeletionTimestamp != nil && !c.isMigrationHandedOffToVirtHandler(migration, vmi) && !migration.IsFinal()) || c.isMigrationAbortedBeforeHandoff(getMigrationKey(migration)) {
+		c.markMigrationAbortedBeforeHandoff(getMigrationKey(migration))
+
+		if targetPodExists {
+			// In this case when the migration has been aborted before handoff and target pod exists, delete target pod
+			err := c.clientset.CoreV1().Pods(pod.Namespace).Delete(context.Background(), pod.Name, v1.DeleteOptions{})
+			if err != nil {
+				return fmt.Errorf("cannot delete pending target pod %s/%s for migration although migration is aborted", pod.Name, pod.Namespace)
+			}
+		}
+
```

Comment:
this logic should be in the same places that handlePendingPodTimeout() is called.

Reply:
Why? Can you explain?

Bear in mind that, handlePendingPodTimeout() is not being called when the migration has phases `MigrationScheduled`, `MigrationPreparingTarget` for example - but we still can abort the migration if the handoff did not occur yet.

The reasoning was that if indeed no handoff occurred there is no reason to continue to handle the different cases, quite the contrary, we should immediately prevent all further logic from occurring since it's no longer relevant.